dipy/align/api.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import nibabel as nib
+
+from imwarp import SymmetricDiffeomorphicRegistration
+from imwarp import DiffeomorphicMap
+from metrics import CCMetric
+import vector_fields as vfu
+
+
+def syn_registration(moving, static, metric=CCMetric(3), 
```

Comment:
This function can go directly in dipy.align.imwarp there is no reason to create a new api.py submodule. Why because then we will always have an ambiguous question to answer. Which is what goes to api.py and what doesn't. This also creates a problem with the standard way of writing scripts which is what we were using so far. 

I would write it in the following way:
This function that you have here would go to dipy.align.imwarp in a bit different form:
```
def syn_registration(static, static_affine, moving, moving_affine, metric='CC', dim=3, level_iters, other_params ...):

Parameters
----------------
static : ndarray
```
Notice here that I am bringing back the affines in the parameters because that is giving us higher flexibility to play with different coordinate transforms.  Also, notice that metric is now just a string and the actual object will be created inside the syn_registration object. There is no need to create this object outside this function. Also, notice that dim is now a new parameter. This is because in most cases we will use this in 3D images so this parameter can be easily set to default of 3 and then just change the metric name.

Okay, so now that we have a very simple function which would run with one static and one moving image and would work smoothly from our standard scripts we can go and see how the workflow interface would be. 

So, in dipy.workflows.align we can have now a function called for example
```
def syn_registration_interface(inputs_static, inputs_moving, outputs_moved, other_params ...):
"""
Parameters
----------------
inputs_static : str (filepath) of File or list of str or list of File
...
"""
# binds all inputs together and creates the output files
```

This function now has the capability to run the syn_registration function which we created before with multiple input files and outputs. So, it connects one or more collections of inputs to one or more outputs. Now, that we have the interface function that can run with multiple files, let's go and create the command line interface in dipy/bin/dipy_syn

```
from dipy.workflow.segment import syn_registration_interface
from dipy.fixes import argparse as arg
parser = arg.ArgumentParser()
parser.add_argument('input_static', action='store')
parser.add_argument('input_moving', action='store')
parser.add_argument('output_directory', action='store')

if __name__ == "__main__":
    params = parser.parse_args()
    syn_registration_interface(params.input_static, params.input_moving, params.output_directory, params.other ...)
```
Now that we have this very thin cmd interface we can go in a terminal and run for example

```
>>> dipy_syn -i Data/exp10/atlas/static.nii.gz, Data/exp10/subj*/moving.nii.gz -o Data/exp10/subj*/moved.nii.gz --metric "CC" -iters ...
```
This is now beautiful because we have a very thin CLI which also is using the power of bash which allows you to have * and ? etc and process multiple subjects at the same time.

Also, we can use the same interface from a website or an ipython notebook too. The CLI is only one option.

How does that sound?